pkg/storage/etcd.go

```diff
@@ -36,37 +37,77 @@ func NewEtcdStorage(endpoints []string) (*EtcdStorage, error) {
 }
 
 // Store stores the key
-func (etcds *EtcdStorage) Store(key string, value Data) error {
+func (etcds *EtcdStorage) StoreCheckin(key string, value time.Time) error {
 	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
-	encoded, _ := json.Marshal(value)
-	_, err := etcds.kv.Put(ctx, fmt.Sprintf("env=%s", key), string(encoded))
+	encoded, err := json.Marshal(value)
+	if err != nil {
+		cancel()
+		return errors.Wrap(err, "failed to store key")
+	}
+	key = strings.TrimSuffix(strings.TrimPrefix(key, "env/"), "/last") // sanitize the strings
+	_, err = etcds.kv.Put(ctx, fmt.Sprintf("env/%s/last", key), string(encoded))
+	cancel()
+	return errors.Wrap(err, "failed to store key")
+}
+
+func (etcds *EtcdStorage) StoreIncident(key string, date string, value []time.Time) error {
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+	encoded, err := json.Marshal(value)
+	if err != nil {
+		cancel()
+		return errors.Wrap(err, "failed to store key")
+	}
+	key = strings.TrimSuffix(strings.TrimPrefix(key, "env/"), "/last") // sanitize the strings
+	_, err = etcds.kv.Put(ctx, fmt.Sprintf("env/%s/incidents/%s", key, date), string(encoded))
 	cancel()
 	return errors.Wrap(err, "failed to store key")
 }
 
 // All returns a map of all keys and values
-func (etcds *EtcdStorage) All() (map[string]Data, error) {
+func (etcds *EtcdStorage) AllCheckins() (map[string]time.Time, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+	v, err := etcds.kv.Get(ctx, "env", clientv3.WithPrefix())
+	cancel()
+	if err != nil {
+		return map[string]time.Time{}, errors.Wrap(err, "failed to retrieve all checkins")
```

Comment:
Is there a reason to return an empty map rather than `nil`?

`nil` would make these returns more straightforward to read